src/main/java/com/fcgl/madrid/points/service/UserActionService.java

```diff
@@ -1,22 +1,70 @@
 package com.fcgl.madrid.points.service;
 
+import com.fcgl.madrid.points.model.Action;
+import com.fcgl.madrid.points.model.Trophy;
 import com.fcgl.madrid.points.model.UserAction;
+import com.fcgl.madrid.points.model.UserTrophy;
+import com.fcgl.madrid.points.repository.ActionRepository;
+import com.fcgl.madrid.points.repository.TrophyRepository;
 import com.fcgl.madrid.points.repository.UserActionRepository;
+import com.fcgl.madrid.points.repository.UserTrophyRepository;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
 
 @Service
 public class UserActionService {
     private UserActionRepository userActionRepository;
+    private ActionRepository actionRepository; // Is this the correct way to access the action table?
+    private TrophyRepository trophyRepository;
+    private UserTrophyRepository userTrophyRepository;
 
     @Autowired
-    public UserActionService(UserActionRepository userActionRepository) {
+    public UserActionService(UserActionRepository userActionRepository,
+                             ActionRepository actionRepository,
+                             TrophyRepository trophyRepository,
+                             UserTrophyRepository userTrophyRepository) {
         this.userActionRepository = userActionRepository;
+        this.actionRepository = actionRepository;
+        this.trophyRepository = trophyRepository;
+        this.userTrophyRepository = userTrophyRepository;
     }
 
     public List<UserAction> findAll() {
         return userActionRepository.findAll();
     }
+
+    public ResponseEntity<Trophy> addUserAction(Long userID, Long actionID) {
+
+        // Check that the action exists
+        Action action = actionRepository.findById(actionID).orElse(null);
+        if (action == null) {
+            return ResponseEntity.badRequest().body(null);
+        }
+
+        UserAction userAction = userActionRepository.findByUserIdAndAction(userID, action).orElse(null);
+        if (userAction == null) {
+            // We haven't seen this action for this user,
+            // Let's create a new entry with count equal to 1
+            userAction = new UserAction(userID, action, 1);
+            userActionRepository.save(userAction);
+        } else {
+            // We have seen this action for this user before
+            // just increment the action counter
+            userAction.setCount(userAction.getCount() + 1);
+            userActionRepository.save(userAction);
+        }
+
+        // TODO: Update user points...
```

Comment:
Do we really need a separate table for this?

Reply:
The alternative being placing the user points where?

Reply:
In the action table. The number of points belonging to an action is part of the action definition. The action title could be very decisive on how many points it should give to a user performing this action. what do you think?

Reply:
The `UserPoint` table keeps track of how many points a user has. User can get points by completing actions or by getting trophies. We don't currently have points in the Action table which is where I think the confusion is coming from. We should add a points column to the action table

Reply:
Does that answer your question?

Reply:
> We should add a points column to the action table

Exactly what I want to do! but why do we need the `ActionPoint` table then? Why was it added?

Reply:
Whoops, I was looking at a local branch that didn't have that table. Haha I suppose you're right. We don't need the `ActionPoint` table, I can take it out in a subsequent PR or you can take it out in this PR

Reply:
I'll do it and add the points column. Will adapt the the end point as well.